### PR TITLE
Add io.mrarm.mcpelauncher client

### PR DESCRIPTION
**Pull request by ChristopherHX**
The actual Mincraft Android gamelauncher.
Cannot upgrade runtime from 5.12
 - the stable launcher for version 0.12 to 1.16 is 32bit only,
 - multilib is complicated in flatpak.
 - x86_64 is experimental and only supports 1.14.60.5+

Downloader https://github.com/flathub/flathub/pull/1594

### CODE REVIEW

**Comment by ChristopherHX** [comment]
bot, build io.mrarm.mcpelauncher-client

**Comment by ChristopherHX** [comment]
bot, build io.mrarm.mcpelauncher-client

File: io.mrarm.mcpelauncher-client.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
+{
+    "id": "io.mrarm.mcpelauncher-client",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.12",
```
Comment (barthalion) [inline io.mrarm.mcpelauncher-client.json:4]: It really should be upgraded. What was the problem with 5.14?

File: io.mrarm.mcpelauncher-client.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "mcpelauncher-client",
+    "build-options" : {
+        "cflags": "-O2 -g -DNDEBUG",
+        "cxxflags": "-O2 -g -DNDEBUG",
+        "config-opts": [ "-DGAMEWINDOW_SYSTEM=GLFW", "-DCMAKE_THREAD_LIBS_INIT=-lpthread", "-DCMAKE_HAVE_THREADS_LIBRARY=1", "-DCMAKE_USE_PTHREADS_INIT=1" ]
```
Comment (barthalion) [inline io.mrarm.mcpelauncher-client.json:10]: This should be defined under specific module.

File: io.mrarm.mcpelauncher-client.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    "only-arches": [ "arm" ]
+                }
+            ],
+            "buildsystem": "cmake-ninja",
+            "only-arches": [ "i386", "arm" ]
+        },
+        {
+            "name": "mcpelauncher-client",
+            "sources": [
+                {
```
Comment (barthalion) [inline io.mrarm.mcpelauncher-client.json:62]: It's happy coincidence it works. SDK ships clang, libtinfo can be built from source as well with desired ABI version.

File: io.mrarm.mcpelauncher-client.json (hunk 1 above)
Comment (ChristopherHX) [reply io.mrarm.mcpelauncher-client.json:4]: With 5.14 the launcher would loose its i386 support, which can run much more versions of Minecraft on Linux (from last years), but specifing 5.12 only for i386 won't seem to work.
I guess only provide mcpelauncher-client32 (i386) in the AppImage not in flatpak, then the new sdk is fine.

File: io.mrarm.mcpelauncher-client.json (hunk 3 above)
Comment (ChristopherHX) [reply io.mrarm.mcpelauncher-client.json:62]: Yes I know, SDK 5.14 ships clang-8, but it's one version too old. One new dependency of the 64bit compatible (experimental) launcher requires clang-9 or newer to compile sucessful.
Probably the upcoming freedesktop sdk 20.08 will have the right clang version in the sdk.
(libtinfo5 is a runtime dependency of the offical clang-9 release page, not referenced anywhere else)
> libtinfo can be built from source

done in my local version now.
So should clang be build from source too?

File: io.mrarm.mcpelauncher-client.json (hunk 2 above)
Comment (ChristopherHX) [reply io.mrarm.mcpelauncher-client.json:10]: Ok

File: io.mrarm.mcpelauncher-client.json (hunk 1 above)
Comment (ChristopherHX) [reply io.mrarm.mcpelauncher-client.json:4]: Old Minecraft Version support for intel / amd.
The x86_64 launcher also needs newer cpus than the x86 32bit version.
There are x86_64 pc's that cannot run Minecraft Android x86_64, but x86.
x86 only needs SSSE3 support
x86_64 needs additionally SSE4.1, SSE4.2 and POPCNT support

File: io.mrarm.mcpelauncher-client.json (hunk 2 above)
Comment (ChristopherHX) [reply io.mrarm.mcpelauncher-client.json:10]: Done

File: io.mrarm.mcpelauncher-client.json (hunk 3 above)
Comment (ChristopherHX) [reply io.mrarm.mcpelauncher-client.json:62]: SDK ships only clang-8, but the project needs **clang-9** or newer.
libtinfo is now build from source, but only a dependency of the standalone clang-9 compiler.
Newer sdk might solve this workaround.

**Comment by ChristopherHX** [comment]
I close this in favour of the first PR #1594, which includes this.

**Pull request closed without merging**